lib/src/config_resolver.rs

```diff
@@ -50,6 +50,8 @@ pub struct ConfigResolutionContext<'a> {
     /// Space-separated subcommand. `jj file show ...` should result in `"file
     /// show"`.
     pub command: Option<&'a str>,
+    /// Hostname from `operation.hostname` config.
+    pub hostname: Option<&'a str>,
```

Comment:
nit: since `hostname` should be available globally, `Option` wouldn't be needed. If the hostname couldn't be fetched, we can use an empty string.